client/document/desktop.jsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import React, { Fragment } from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { jsonStringifyForHtml } from '../../server/sanitize';
+import Head from '../components/head';
+import getStylesheet from './utils/stylesheet';
+
+class Desktop extends React.Component {
+	render() {
+		const {
+			app,
+			faviconURL,
+			i18nLocaleScript,
+			isRTL,
+			lang,
+			urls,
+			hasSecondary,
+			config,
+			isFluidWidth,
+			env,
+			isDebug,
+			badge,
+			abTestHelper,
+			branchName,
+			commitChecksum,
+			devDocs,
+			devDocsURL,
+			feedbackURL,
+		} = this.props;
+		return (
+			<html
+				lang={ lang }
+				dir={ isRTL ? 'rtl' : 'ltr' }
+				className={ classNames( 'is-desktop', { 'is-fluid-width': isFluidWidth } ) }
+			>
+				<Head title="WordPress.com" faviconURL={ faviconURL } cdn={ '//s1.wp.com' }>
+					<link
+						rel="stylesheet"
+						id="main-css"
+						href={
+							urls[ getStylesheet( { rtl: !! isRTL, debug: isDebug || env === 'development' } ) ]
+						}
+						type="text/css"
+					/>
+					<link rel="stylesheet" id="desktop-css" href="/desktop/wordpress-desktop.css" />
+				</Head>
+				<body className={ classNames( { rtl: isRTL } ) }>
+					{ /* eslint-disable wpcalypso/jsx-classname-namespace, react/no-danger */ }
+					<div id="wpcom" className="wpcom-site">
+						<div className="layout">
+							<div className="masterbar" />
+							<div className="layout__content">
+								<div className="wpcom-site__logo noticon noticon-wordpress" />
+								{ hasSecondary && (
+									<Fragment>
+										<div className="layout__secondary" />
+										<ul className="sidebar" />
+									</Fragment>
+								) }
+							</div>
+						</div>
+					</div>
+					{ badge && (
+						<div className="environment-badge">
```

Comment:
Would be nice to turn the environment badge into a re-usable component, so we can use it both here and in the `index` template...

Reply:
I definitely agree 👍 do you mind if I add this with another PR?

Reply:
Another PR is fine!